app/models/etl/metrics.rb

```diff
@@ -14,15 +14,21 @@ def process
   def create_metrics
     Facts::Metric.where(dimensions_date: date).delete_all
 
-    Dimensions::Item.where(latest: true).find_in_batches(batch_size: 50000) do |batch|
-      values = batch.pluck(:id, :organisation_id)
-      metrics = values.map do |value|
+    Dimensions::Item.select(:id, :organisation_id, :link).where(latest: true).find_in_batches(batch_size: 500) do |batch|
```

Comment:
I think we should have an independent `ETL::GAMetrics` that populates the Facts table with GA Metrics and has its own trigger (`rake` task). It should happen the same with Content Quality Metrics, as both of them should be able to run independently.

If GA is down, or having problems, or if we want to run the process for a particular day, we don't have to recreate the whole Facts Table for that day. In particular, when we are deleting the table first.

Reply:
To make it more robust, and regardless of having it split to its own process, I would create Jobs so we guarantee that a failure in one request does not prevent the whole process from working as expected.